modules_core/com.smf.securewebservices/src/com/smf/securewebservices/utils/SecureDataToJson.java

```diff
@@ -68,6 +68,12 @@ public JSONObject toJsonObject(BaseOBObject bob, DataResolvingMode dataResolving
 			final boolean isDerivedReadable = OBContext.getOBContext().getEntityAccessChecker()
 					.isDerivedReadable(bob.getEntity());
 
+			// validate if this entity is readable.
+			if (OBContext.getOBContext().isInAdministratorMode()) {
```

Comment:
If this "toJsonObject" is called, and having an AdministratorMode, then it's going to lose that AdministratorMode, is this the expected behaviour?

public void myClass() {
  try {
    OBContext.getOBContext.setAdminMode();
    //do things
    SecureDataToJson(x, y, z);
    //Do things that need admin mode -> Fails?
  } finally {
    OBContext.restorePreviousMode()   // Fails?
  }
}

Reply:
as far as I could test, it is not failing. would be a good thing to set adminmode again once the call to SecureDataToJson has finished?

Reply:
**update**

tested, and in the `finally` clause, the context is **NOT** in admin mode. but it doesn't seem to be a problem to use the `OBContext.restorePreviousMode()` when the context is not in adminmode. I don't see any "unbalanced calls" warning, nor errors on the behaviour of the process. I think I could even don't check if it is in adminmode to use `OBContext.restorePreviousMode()`

how do I proceed?

Reply:
Maybe if the context was in adminMode before, would be good to take it back to adminmode. Or add a disclaimer/warning/something in the javadoc of **_SecureDataToJson_** letting the user know that the AdminMode will be deactivated.

@Matias-Bernal @RomanMagnoli @sebastianbarrozo @valenvivaldi what do u think?

Reply:
* I would not touch anymore the method that removes the admin mode. I prefer to have the warning in the javadoc that removes it. Whoever uses it, should consider that.